openedx/features/effort_estimation/block_transformers.py

```diff
@@ -0,0 +1,239 @@
+                # technically make our estimate less accurate, especially as we combine these values in parents.
+                # But easier to present a simple integer to any consumers, and precise to-the-second accuracy on our
+                # estimate is not a primary goal.
+                block_structure.override_xblock_field(block_key, self.EFFORT_TIME, math.ceil(time))
+
+            if activities is not None:
+                block_structure.override_xblock_field(block_key, self.EFFORT_ACTIVITIES, activities)
+
+        # Get bucket for this experiment. 0 is no estimate. 1 is only on sections. 2 is only on subsections.
+        # For cleanup ticket AA-659: remove everything below.
+
+        # We only want to get the bucket if there is data available - i.e. there is something to actually experiment
+        # on. This helps avoid rollout issues where we don't want to claim a user is in bucket 1 if we haven't even
+        # re-published the course so that it has any estimation data available.
+        root_key = block_structure.root_block_usage_key
+        total_activities = block_structure.get_xblock_field(root_key, self.EFFORT_ACTIVITIES)
+        total_time = block_structure.get_xblock_field(root_key, self.EFFORT_TIME)
+        if not total_activities and not total_time:
+            return
+
+        # Second pass to clear out collected estimate on levels we don't want to share. Just an easy way to test
+        # estimates at different levels, per the experiment.
+        bucket = EFFORT_ESTIMATION_LOCATION_FLAG.get_bucket(course_key=usage_info.course_key)
+        for block_key in block_structure.post_order_traversal():
+            category = block_structure.get_xblock_field(block_key, 'category')
+            allowed = (bucket == 1 and category == 'chapter') or (bucket == 2 and category == 'sequential')
+            if not allowed:
+                block_structure.override_xblock_field(block_key, self.EFFORT_TIME, None)
+                block_structure.override_xblock_field(block_key, self.EFFORT_ACTIVITIES, None)
+
+    @cached_property
+    def _is_on_mobile(self):
+        """Returns whether the current request is from our mobile app."""
+        request = crum.get_current_request()
+        return request and is_request_from_mobile_app(request)
+
+    def _gather_child_values(self, block_structure, block_key, field, default=0):
+        """Collects and sums all child values for field."""
+        return sum([
+            block_structure.get_xblock_field(child_key, field, default=default)
+            for child_key in block_structure.get_children(block_key)
+        ])
+
+    def _estimate_children_effort(self, _usage_info, block_structure, block_key):
+        """Collects time and activity counts for children."""
+        time = self._gather_child_values(block_structure, block_key, self.EFFORT_TIME)
+        time = time or None  # avoid claiming anything takes 0 seconds by coercing to None (no estimate) instead
```

Comment:
nit Why not just return None from the children?

Reply:
I don't think it ended up being a practical difference - some versions of my PR might have treated them differently, can't remember. But mostly because I felt it was more appropriate to say "unknown" than "zero" if we had no time info? Very minor thing.